src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,263 @@
+            return true;}
+        } else return false;
+
     }
 
     public boolean remove(Object o) {
-        return false;
+
+        int index = o.hashCode() % buckets.length;
+        Node bucket = buckets[index];
+        boolean flag = false;
+
+
+        if (bucket == null) {
+            throw new NoSuchElementException("No Element Found");
+
+        }
+
+        if (bucket.data.equals(o)) {
+            buckets[index] = bucket.next;
+            size--;
+            flag = true;
+        }
+        return flag;
     }
 
     public boolean containsAll(Collection<?> c) {
-        return false;
+        for (Object elem : c) {
+            if (!this.contains(elem)) {
+                return false;
+            }
+        }
+        return true;
     }
 
     public boolean addAll(Collection<? extends T> c) {
-        return false;
+        for (T elem : c) {
+            this.add(elem);
+        }
+        return true;
+
     }
 
     public boolean retainAll(Collection<?> c) {
+        for (Object elem : c) {
+            if (this.contains(elem) && elem != null) {
+                this.remove(elem);
+            }
+        }
         return false;
     }
 
     public boolean removeAll(Collection<?> c) {
+        for (Object elem : c) {
+            this.remove(elem);
+        }
         return false;
     }
 
     public void clear() {
+        Node[] newBuckets = new Node[size];
+        buckets = newBuckets;
+
+    }
+
+    public String toString() {
+        return null;
+    }
+
+    static Range<Integer> of(int e1, int e2) {
+        int start = e1;
+        int end = e2;
+        int range = end - start;
 
+        Range<Integer> newRange;
+        if (range == 0) {
+            return new Range<Integer>(0);
+        } else {
+            newRange = new Range<Integer>();
+            for (int i = start; i <= end; i++) {
+                newRange.add(i);
+            }
+        }
+        return newRange;
     }
+
+    static Range<Float> of(float e1, float e2) {
+        float start = e1;
+        float end = e2;
+        float range = end - start;
+
+        Range<Float> newRange;
+        if (range == 0) {
+            return new Range<Float>(0);
+        } else {
+            newRange = new Range<Float>();
+            for (float i = start; i <= end; i+=0.1) {
+                newRange.add(i);
+            }
+        }
+        return newRange;
+    }
+
+    static Range<Double> of(double e1, double e2) {
+        double start = e1;
+        double end = e2;
+        double range = end - start;
+
+        Range<Double> newRange;
+        if (range == 0) {
+            return new Range<Double>(0);
+        } else {
+            newRange = new Range<Double>();
+            for (double i = start; i <= end; i+=0.1) {
+                newRange.add(i);
+            }
+        }
+        return newRange;
+    }
+
+    static Range<Character> of(Character e1, Character e2,Function<Character,Character> function) {
```

Comment:
1. Think about **custom** realization of **Number**. 
2. Try to avoid duplication of code. Each overloaded method **of** contains similar code.